controller/src/main/java/org/jboss/as/controller/AttributeMarshaller.java

```diff
@@ -105,6 +105,33 @@ protected String asString(ModelNode value) {
         }
     }
 
+    private static class ObjectMarshaller extends DefaultAttributeMarshaller {
+        private final boolean marshallSimpleTypeAsAttribute;
+
+        private ObjectMarshaller(boolean marshallSimpleTypeAsAttribute) {
+            this.marshallSimpleTypeAsAttribute = marshallSimpleTypeAsAttribute;
+        }
+
+        @Override
+        public void marshallAsElement(AttributeDefinition attribute, ModelNode resourceModel, boolean marshallDefault, XMLStreamWriter writer) throws XMLStreamException {
+            if (resourceModel.hasDefined(attribute.getName()) && attribute instanceof ObjectTypeAttributeDefinition) {
```

Comment:
Should there be an assert here re: the type of 'attribute'? Documented in the javadoc for the constants?

Perhaps this class and the constants should move to ObjectTypeAttributeDefinition, since the impls are tightly coupled. The ListMarshaller class in AttributeMarshaller isn't tied that way. It probably wouldn't be used except with some variant of ListAttributeDefinition, but it could be.